### PR TITLE
Document some register attributes

### PR DESCRIPTION
**Pull request by dgilge**
Addressing several attributes of `register` for use with third-party apps:
1. `app`
2. `excluded_fields`
3. Reference for other attributes

I needed this and searched for a clean solution (almost opened an issue) and was happy when I saw that it is implemented already. People should know that all these features exist! :)

I implemented it in my project and it works. I only changed the documetation. Therefore I assume that I don't have to write a test.

### CODE REVIEW

File: docs/advanced.rst [hunk 1]
```diff
@@ -428,4 +445,4 @@ This may be useful when you want to construct timelines and need to get only the
     new_record, old_record = p.history.all()
     delta = new_record.diff_against(old_record)
     for change in delta.changes:
-        print("{} changed from {} to {}".format(change.field, change.old, change.new))
\ No newline at end of file
+        print("{} changed from {} to {}".format(change.field, change.old, change.new))
```
Comment (rossmechanic) [inline docs/advanced.rst]: I don't see a diff here? Spacing?

**Comment by rossmechanic** [comment]
Thanks for updating the docs! Just one comment – looks like there was no change (spacing?). Might want to just get rid of that and then I'll merge.

File: docs/advanced.rst (hunk 1 above)
Comment (dgilge) [reply docs/advanced.rst]: GitHub adds a blank line at the end. I'm going to fix it.

Comment (dgilge) [reply docs/advanced.rst]: No, it doesn't. I'm afraid I don't know how to fix this.

Comment (rossmechanic) [reply docs/advanced.rst]: No worries. Merging. Thanks!

**Pull request merged**